postgres/functions/get_chw_last_sync.sql

```diff
@@ -0,0 +1,42 @@
+CREATE OR REPLACE FUNCTION public.get_chw_last_sync() RETURNS TABLE(partner text, chw_id text, chw_username text, last_sync date) AS $$
+DECLARE partners cursor IS (
+        SELECT DISTINCT ON (partner_name)
+            partner_name AS name,
+            port
+        FROM impactconfig
+        WHERE close_date IS NULL
+    );
+DECLARE credentials record;
+BEGIN 
+    SELECT value->>'user' AS user, value->>'password' AS password FROM configuration WHERE KEY = 'dblink' INTO credentials;
+    FOR partner IN partners LOOP RETURN query
+    SELECT
+        *
+    FROM dblink(
+            FORMAT(
+                'dbname=%s host=localhost port=%s user=%s password=%s',
+                partner.name,
+                partner.port,
+                credentials.user,
+                credentials.password
+            ),
+            '
+            SELECT
+                current_database() as partner,
+                form.chw as chw_id,
+                c_user.doc->>''name'' as chw_username,
+                max(c_info.doc->>''latest_replication_date'')::date as last_sync
+            FROM
+                form_metadata form
```

Comment:
I'm curious what your thoughts are on this @freddieptf

Reply:
we can use the same approach and get the data from `contactview_metadata`

Reply:
@derickl I don't think it's possible to do this reliably for multiple projects using `contactview_metadata`. We have varying hierarchies for projects and the names assigned to contact types also change project to project. Another option would be using `chw_hierarchy` to get all children contacts under the chw's place but we'd have to make assumptions on how deep the hierarchy is under the chw's place and it would increase complexity. Considering how often contacts are created it might not really be necessary for this function to work as expected.

Reply:
All we need to show is that it's a contact. Chances are only contacts created offline would have a replication date. The first couple of months in a project would have a lot of contact creation.

This might be worth a conversation. You could open a separate issue to explore this improvement

Reply:
https://github.com/medic/cht-app-monitoring-data-ingest/issues/92